### PR TITLE
refactor(drawer): add default value to DrawerSidebarContext - FE-4224

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
Adding default value to DrawerSidebarContext will allow us to use object destructuring

### Current behaviour
Default value is missing

### Checklist

- [x] Commits follow our style guide
- [ ] <del> Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [ ] <del> Unit tests added or updated if required
- [ ] <del> Cypress automation tests added or updated if required
- [ ] <del> Storybook added or updated if required
- [ ] <del> Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### Testing instructions
I think we should just check all components that use DrawerSidebarContext and check if there are any changes. No changes should be visible. 
Components that uses DrawerSidebarContext:
- Drawer component
- Flat Table component
- Tabs

### CODE REVIEW

File: src/components/tabs/tabs.component.js [hunk 1]
```diff
@@ -248,10 +247,10 @@ const Tabs = ({
     return (
       <TabsHeader
         align={align}
-        position={sidebarContext ? "left" : position}
+        position={isInSidebar ? "left" : position}
         role="tablist"
         extendedLine={extendedLine}
-        alternateStyling={variant === "alternate" || !!sidebarContext}
+        alternateStyling={variant === "alternate" || !!isInSidebar}
```
Comment (edleeks87) [inline src/components/tabs/tabs.component.js:253]:
```suggestion
        alternateStyling={variant === "alternate" || isInSidebar}
```
It's already a boolean now

File: src/components/tabs/tabs.spec.js [hunk 2]
```diff
@@ -235,6 +235,8 @@ describe("Tabs", () => {
   describe('When "selectedTabId" is passed a valid "tabId"', () => {
     it("displays the specified Tab", () => {
       const wrapper = render({ selectedTabId: "uniqueid2" });
+
+      // console.log(wrapper.find(Tab).debug())
```
Comment (edleeks87) [inline src/components/tabs/tabs.spec.js:239]: commented code

**Review by edleeks87 — COMMENTED** [review]
I think it's worth updating the other components that use `DrawerSidebarContext`:

- In `FlatTable`
https://github.com/Sage/carbon/blob/531d7dec396c9781517ddb4429bdc58cfd83338c/src/components/flat-table/flat-table.component.js#L46
can now just destructure `isInSidebar` like so `{({isInSidebar}) => (`

https://github.com/Sage/carbon/blob/531d7dec396c9781517ddb4429bdc58cfd83338c/src/components/flat-table/flat-table.component.js#L55
can just be `isInSidebar={isInSidebar}`

- In `FlatTableRow` you can do the same
https://github.com/Sage/carbon/blob/531d7dec396c9781517ddb4429bdc58cfd83338c/src/components/flat-table/flat-table-row/flat-table-row.component.js

File: src/components/tabs/tabs.spec.js (hunk 2 above)
Comment (dawidzarzycki) [reply src/components/tabs/tabs.spec.js:239]: wops :3

**Review by edleeks87 — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

**Pull request merged**